pl_bolts/models/self_supervised/fixmatch/comatch_module.py

```diff
@@ -0,0 +1,197 @@
+import torch
+from pytorch_lightning.utilities.cli import LightningCLI
+
+from pl_bolts.models.self_supervised.fixmatch.fixmatch_module import FixMatch, WideResnet
+
+
+class Queue:
+    def __init__(self, value, size):
+        self.value = value
+        self.size = size
+        self.ptr = 0
+
+    def __call__(self):
+        return self.value
+
+    def update(self, new_value, total_batch_size):
+        if new_value.device != self.value.device:
+            # Move to the updated device.
+            self.value = self.value.to(new_value.device)
+        self.value[self.ptr : self.ptr + total_batch_size, :] = new_value
+        self.ptr = (self.ptr + total_batch_size) % self.size
+
+
+class CoMatch(FixMatch):
+    def __init__(
```

Comment:
can we add docstrings with basic descriptions and references?
Also pls add docstring for init so when it is called with PL CLI there are argument explanations